pandas/_libs/algos.pyx

```diff
@@ -393,6 +393,81 @@ def nancorr_spearman(ndarray[float64_t, ndim=2] mat, Py_ssize_t minp=1) -> ndarr
     return result
 
 
+# ----------------------------------------------------------------------
+# Kendall correlation
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def nancorr_kendall(ndarray[float64_t, ndim=2] mat, Py_ssize_t minp=1) -> ndarray:
```

Comment:
actually, can you add a small doc-string with Parameters and Returns (esp shapes / dtypes) & put a link the kendall algo (wikipedia fine)